go-controller/pkg/ovn/pods.go

```diff
@@ -46,16 +48,17 @@ func (oc *DefaultNetworkController) syncPods(pods []interface{}) error {
 				return err
 			}
 		} else if oc.isPodScheduledinLocalZone(pod) {
-			expectedLogicalPortName, annotations, err = oc.allocateSyncPodsIPs(pod)
-			if err != nil {
-				return err
+			if switchesNotFound[pod.Spec.NodeName] {
+				klog.Warningf("Cannot allocate IPs for %s/%s, node was not found after 30 seconds", pod.Namespace, pod.Name)
+				continue
 			}
-		} else {
```

Comment:
you removed some logic here that shouldnt have been removed, like skipping non local zone pods, and checking if annotations is nil. This is causing the unit tests to crash. Also you need to check if there is an error other than nodeNotFoundError:

```
[trozet@fedora go-controller]$ git diff
diff --git a/go-controller/pkg/ovn/pods.go b/go-controller/pkg/ovn/pods.go
index 7c23fba85..7d53f2ecf 100644
--- a/go-controller/pkg/ovn/pods.go
+++ b/go-controller/pkg/ovn/pods.go
@@ -54,11 +54,20 @@ func (oc *DefaultNetworkController) syncPods(pods []interface{}) error {
                        }
                        expectedLogicalPortName, annotations, err = oc.allocateSyncPodsIPs(pod)
 
-                       if errors.Is(err, nodeNotFoundError) {
-                               klog.Warningf("Cannot allocate IPs for %s/%s, node was not found after 30 seconds", pod.Namespace, pod.Name)
-                               switchesNotFound[pod.Spec.NodeName] = true
-                               continue
+                       if err != nil {
+                               if errors.Is(err, nodeNotFoundError) {
+                                       klog.Warningf("Cannot allocate IPs for %s/%s, node was not found after 30 seconds", pod.Namespace, pod.Name)
+                                       switchesNotFound[pod.Spec.NodeName] = true
+                                       continue
+                               }
+                               return err
                        }
+               } else {
+                       continue
+               }
+
+               if annotations == nil {
+                       continue
                }

```